packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -50,6 +52,22 @@ export abstract class AbstractResourcePreferenceProvider implements PreferencePr
         return this.preferences;
     }
 
+    async setPreference(key: string, value: string): Promise<void> {
+        const resource = await this.resource;
+        if (resource.saveContents) {
+            let preferences = this.preferences;
+            if (preferences) {
+                preferences[key] = value;
+            } else {
+                preferences = {[key]: value};
+            }
+            const content = await resource.readContents();
+            const indent = detectIndent(content).ident || '   ';
+            await resource.saveContents(JSON.stringify(preferences, undefined, indent));
```

Comment:
have you looked into the `modify` option of `jsonc-parser`?

Reply:
Yes I have, `JSON.stringify` looks more suitable according to given parameters.

Reply:
@vinokurig, what exactly are the given parameters here?

I think it it worth a try to [modify](https://github.com/Microsoft/node-jsonc-parser/blob/master/src/main.ts#L331) with `jsonc-parser`. 

- you won't have to detect indent
- it's already there
- and you're not going to delete users comments

have you seen, that there are potentially comments in that json file? see https://github.com/vinokurig/theia/blob/47b2bca56e3a5ba5861fcce79a17eb1047984da7/packages/preferences/src/browser/abstract-resource-preference-provider.ts#L73

Reply:
done